File: BadApple-ASCII/backend.go
package main

import (
	"io/ioutil"
	"log"
	"os/exec"
	"path/filepath"
	"regexp"
	"sort"
	"strconv"
	"sync"
)

func extractNumericPart(filename string) int {
	re := regexp.MustCompile(`\d+`)
	match := re.FindString(filename)
	if match == "" {
		return int(^uint(0) >> 1) // Max int value
	}
	num, _ := strconv.Atoi(match)
	return num
}

func processFile(filePath string, wg *sync.WaitGroup) {
	defer wg.Done()
	cmd := exec.Command("ascii-image-converter.exe", filePath)
	err := cmd.Run()
	if err != nil {
		log.Printf("Failed to process file %s: %v", filePath, err)
	}
}

func openFilesInOrder(directory string) {
	files, err := ioutil.ReadDir(directory)
	if err != nil {
		log.Fatalf("Failed to read directory: %v", err)
	}

	var pngFiles []string
	for _, file := range files {
		if filepath.Ext(file.Name()) == ".png" {
			pngFiles = append(pngFiles, file.Name())
		}
	}

	sort.Slice(pngFiles, func(i, j int) bool {
		return extractNumericPart(pngFiles[i]) < extractNumericPart(pngFiles[j])
	})

	var wg sync.WaitGroup
	for _, file := range pngFiles {
		wg.Add(1)
		go processFile(filepath.Join(directory, file), &wg)
		wg.Wait()
	}

	wg.Wait()
}

func main() {
	directoryPath := `C:\Users\Hp\Documents\workspace\BadApple\Frames-1`
	openFilesInOrder(directoryPath)
}

File: BadApple-ASCII/backend.py
import os
import time
import subprocess
import re
import sys

def extract_numeric_part(filename):
    match = re.search(r'\d+', filename)
    return int(match.group()) if match else float('inf')

def open_files_in_order(directory):
    files = sorted(
        (file for file in os.listdir(directory) if file.endswith(".png")),
        key=extract_numeric_part
    )

    for file in files:
        print('-' * 116)
        file_path = os.path.join(directory, file)

        subprocess.run(["ascii-image-converter.exe", file_path])

        time.sleep(1 / 10)
        


    sys.stdout.write("\033[K\r")
    sys.stdout.flush()
    

directory_path = r"C:\Users\Hp\Documents\workspace\youtube-dl\BadApple"
open_files_in_order(directory_path)
